src/explore-education-statistics-admin/src/components/editable/EditableContentBlock.tsx

```diff
@@ -120,6 +125,43 @@ const EditableContentBlock = ({
     };
   }, [comments, transformImageAttributes]);
 
+  // Check content for images without alt text.
+  useEffect(() => {
+    if (content && !isEditing) {
+      const parsedContent = parseHtmlString(content);
+
+      const parsedContentArray = Array.isArray(parsedContent)
+        ? parsedContent
+        : [parsedContent];
+
+      const hasInvalidImage = parsedContentArray.some(element => {
+        if (typeof element === 'string') {
+          return false;
+        }
+        if (element.type === 'img' && !element.props.alt) {
+          return true;
+        }
+
+        if (element.props.children) {
+          const childrenArray = Array.isArray(element.props.children)
+            ? element.props.children
+            : [element.props.children];
+
+          return childrenArray.some(
+            (child: JSX.Element | string) =>
+              typeof child !== 'string' &&
+              child.type === 'img' &&
+              !child.props.alt,
+          );
+        }
+
+        return false;
+      });
```

Comment:
A much simpler way to do this is to use the DOM e.g.

```ts
const hasInvalidImage = !!contentRef.current?.querySelector(
  'img:not([alt]), img[alt=""]',
);

toggleAltTextMessage(hasInvalidImage);
```

This just needs a `contentRef` setup somewhere e.g. on the `inert` wrapper div:
```tsx
const contentRef = useRef<HTMLDivElement>(null);

<div inert="" ref={contentRef}>
  <ContentHtml ... />
</div>
```

Alternatively, could do it by adding using a `forwardRef` on `ContentHtml`